File: backend-java/src/main/java/com/backend/java/backendjava/usercases/CardUserCase.java
package com.backend.java.backendjava.usercases;

import com.backend.java.backendjava.adapters.CardAdapter;
import com.backend.java.backendjava.dtos.requests.CardRequest;
import com.backend.java.backendjava.dtos.responses.CardResponse;
import com.backend.java.backendjava.entities.Card;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class CardUserCase implements CardAdapter {
    public CardResponse createCard(CardRequest request) {
        validateCard(request);
        return  null;
    }

    public Page<CardResponse> readCard() {
        return null;
    }

    public CardResponse updateCardById(Long id, CardRequest request) {
        validateCard(request);
        return null;
    }

    public void deleteCardById(Long id) {}

    private void  validateCard(CardRequest request) {
        if (request.getCardValue() == 0) {
            throw  new RuntimeException("value card 0");
        }
    }
}

File: backend-kotlin/src/main/kotlin/com/backend/kotlin/backendkotlin/usercases/CardUserCase.kt
package com.backend.kotlin.backendkotlin.usercases

import com.backend.kotlin.backendkotlin.adapters.CardAdapter
import com.backend.kotlin.backendkotlin.dtos.requests.CardRequest
import com.backend.kotlin.backendkotlin.dtos.responses.CardResponse
import org.springframework.data.domain.Page
import org.springframework.stereotype.Component


@Component
class CardUserCase : CardAdapter {
    override fun createCard(request: CardRequest?): CardResponse? {
        validateCard(request)
        return null
    }

    override fun readCard(): Page<CardResponse?>? {
        return null
    }

    override fun updateCardById(id: Long?, request: CardRequest?): CardResponse? {
        validateCard(request)
        return null
    }

    override fun deleteCardById(id: Long?) {}

    private fun validateCard(request: CardRequest?) {
        if (request?.cardValue?.toInt() == 0) {
            throw RuntimeException("value card 0")
        }
    }
}